cmd/pilot-agent/main.go

```diff
@@ -47,60 +56,87 @@ var (
 		Short: "Envoy proxy agent",
 		RunE: func(c *cobra.Command, args []string) error {
 			// set values from environment variables
-			if sidecar.IPAddress == "" {
-				sidecar.IPAddress = os.Getenv("INSTANCE_IP")
-			}
-			if sidecar.ID == "" {
-				sidecar.ID = os.Getenv("POD_NAME") + "." + os.Getenv("POD_NAMESPACE")
-			}
-			// receive mesh configuration
-			mesh, err := cmd.ReadMeshConfig(meshconfig)
-			if err != nil {
-				return multierror.Prefix(err, "failed to read mesh configuration.")
-			}
+			if serviceregistry == proxy.KubernetesRegistry {
```

Comment:
@costinm please keep in mind that until istio's config server is integrated, the logic in discovery and agent's main.go where an adapter is chosen is completely subject to change.  Ideally, yes, the config pilot retrieves from the config server on start will have fields shared between service registry implementations